pages/api/user/index.ts

```diff
@@ -0,0 +1,180 @@
+  res: NextApiResponse<ApiResponse<EmptyPayload>>,
+) {
+  const isExisting = (await getUserIfExists(currentUid)) !== null;
+  if (isExisting) {
+    return res
+      .status(HttpStatus.CONFLICT)
+      .json(createJsonResponse({}, { type: ERROR, message: `UID ${currentUid} already exists` }));
+  }
+
+  await prisma.user.create({ data: { uid: currentUid } });
+  return res
+    .status(HttpStatus.OK)
+    .json(createJsonResponse({}, { type: SUCCESS, message: `New user with UID ${currentUid} added.` }));
+}
+
+async function handlePostWithOldToken(
+  currentUid: string,
+  oldUserToken: string,
+  req: NextApiRequest,
+  res: NextApiResponse<ApiResponse<EmptyPayload>>,
+) {
+  const currentIsAnonymous = (await getUserFromJwt(req.headers.authorization as string)).isAnonymous;
+  if (currentIsAnonymous) {
+    return res
+      .status(HttpStatus.UNAUTHORIZED)
+      .json(createJsonResponse({}, { type: ERROR, message: 'An unverified account cannot be the target of link.' }));
+  }
+
+  let oldUserDetails: UserDetailsFromRequest;
+  try {
+    oldUserDetails = await getUserFromJwt(oldUserToken);
+  } catch (error) {
+    return res
+      .status(HttpStatus.UNAUTHORIZED)
+      .json(createJsonResponse({}, { type: ERROR, message: `Could not validate and decode 'old_token'.` }));
+  }
+
+  const oldUid = oldUserDetails.uid;
+  if (!oldUserDetails.isAnonymous) {
+    return res
+      .status(HttpStatus.BAD_REQUEST)
+      .json(createJsonResponse({}, { type: ERROR, message: `UID ${oldUid} is already linked` }));
+  }
+
+  const result = await linkAccount(oldUid, currentUid);
+
+  switch (result) {
+    case LinkAccountResult.OLD_USER_DELETED:
+      return res.status(HttpStatus.OK).json(
+        createJsonResponse(
+          {},
+          {
+            type: INFORMATION,
+            message: `Old UID ${oldUid} is no longer in database. New user ${currentUid} has been linked.`,
+          },
+        ),
+      );
+
+    case LinkAccountResult.NEW_USER_HAS_DATA:
+      return res
+        .status(HttpStatus.CONFLICT)
+        .json(createJsonResponse({}, { type: ERROR, message: 'New UID already has user data and cannot be linked.' }));
+
+    case LinkAccountResult.NEW_USER_LINKED:
+      return res
+        .status(HttpStatus.OK)
+        .json(
+          createJsonResponse(
+            {},
+            { type: SUCCESS, message: `Old UID ${oldUid} has been replaced with new UID ${currentUid}.` },
+          ),
+        );
+  }
+}
+
+async function handleDelete(uid: string, _req: NextApiRequest, res: NextApiResponse<ApiResponse<EmptyPayload>>) {
+  // UID is a primary key, so the count will either be 0 (UID does not exist in database) or 1 (successful delete).
+  const user = await getUserIfExists(uid);
+  if (!user) {
+    return res
+      .status(HttpStatus.NOT_FOUND)
+      .json(createJsonResponse({}, { type: ERROR, message: `Could not find UID ${uid}.` }));
+  }
+
+  await prisma.application.deleteMany({ where: { userId: uid } });
```

Comment:
If I am not wrong, applications will be deleted when the user is deleted due to the `onDelete: Cascade` constraint?

Reply:
My mistake, I thought it didn't.